### PR TITLE
fix #195 - add uniqueness to option name by question

### LINKED ISSUE #195: add uniqueness to option name by question
The following scenario is invalid
`Question A` has two options: `option 1`  and `option 1`
`option 1` is repeated for same question

The following scenario is valid
`Question A` has two options: `option 1` and `option 2`
`Question B` has two options: `option 1` and `option 2`
`option 1` is repeated for different questions

### PR DESCRIPTION
**Pull request by danielnottingham**
Screenshot from 2021-06-05 11-21-09

fix #195

### CODE REVIEW

File: backend/spec/models/option_spec.rb [hunk 1]
```diff
@@ -22,11 +22,17 @@
     end
 
     describe 'when many options are set to correct' do
-      it { expect(option.errors.full_messages).to match(['Single choice questions should have no more than one correct option.']) }
+      it { expect(option.errors.full_messages).to match(['Name has already been taken', 'Single choice questions should have no more than one correct option.']) }
```
Comment (edimossilva) [inline backend/spec/models/option_spec.rb:25]: this spec is not related to the issue that you are working on, maybe you should change the option name, so this spec wouldn't change

File: backend/spec/models/option_spec.rb [hunk 2]
```diff
@@ -22,11 +22,17 @@
...
       it { expect(option).not_to be_valid }
     end
   end
 
+  describe 'uniqueness' do
+    before { create(:option) }
```
Comment (edimossilva) [inline backend/spec/models/option_spec.rb:31]: is this necessary?

File: backend/spec/models/question_spec.rb [hunk 3]
```diff
@@ -25,7 +25,9 @@
 
   describe 'when updating to single choice with multiple correct options.' do
     before do
-      create_list(:option, 3, correct: true, question_id: question_multiple.id)
+      create(:option, name: 'option_one', correct: true, question_id: question_multiple.id)
+      create(:option, name: 'option_two', correct: true, question_id: question_multiple.id)
+      create(:option, name: 'another_option', correct: true, question_id: question_multiple.id)
```
Comment (edimossilva) [inline backend/spec/models/question_spec.rb:30]: if you are doing all these changes, maybe you should change the option factory

**Review by edimossilva — CHANGES_REQUESTED** [review]
@danielnottingham, nice, just some small adjusts!

File: backend/spec/models/option_spec.rb (hunk 2 above)
Comment (danielnottingham) [reply backend/spec/models/option_spec.rb:31]: Yes, it's necessary. Because we need to have an option created to test this validation. And in option_spec there is only one let!(build for option).

**Review by edimossilva — APPROVED** [review]
LGTM! Nice!
Squash and merge
